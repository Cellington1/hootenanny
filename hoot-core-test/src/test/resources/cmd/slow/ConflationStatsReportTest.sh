#!/bin/bash
set -e

mkdir -p test-output/cmd/csr

# PRE TEST FILE GENERATION 

hoot conflate -D writer.include.debug=true -D stats.class="hoot::ScriptStatsComposer" -D  stats.script=./report/csr-default.js -D stats.format=asciidoc -D stats.output=test-output/cmd/csr/csr ./test-files/conflate/unified/AllDataTypesA.osm ./test-files/conflate/unified/AllDataTypesB.osm test-output/cmd/csr/csr.osm --stats >& /dev/null

# TEST 1: integrity check of the output osm file when enabling Report Generation and stats
#  - will produce difference which will make different from baseline
hoot conflate -D writer.include.debug=true -D stats.class="hoot::ScriptStatsComposer" -D  stats.script=./report/csr-default.js -D stats.format=html -D stats.output=test-output/cmd/csr/csr ./test-files/ToyTestA.osm ./test-files/ToyTestB.osm test-output/cmd/csr/csr2.osm --stats >& /dev/null
hoot --is-match test-output/cmd/csr/csr.osm test-files/cmd/slow/csr/csr.osm || diff test-output/cmd/csr/csr.osm test-files/cmd/slow/csr/csr.osm 
echo "completed test 1"

# TEST 2: simple file test to make sure asciidoc created
FILE_ASCIIDOC=test-output/cmd/csr/csr.asciidoc
if [ ! -f $FILE_ASCIIDOC ]; then
  echo "ConflationStatsReportTest: asciidoc file not generated"
else
  echo "passed test 2"
fi

# TEST 3: simple file test to make sure html created
FILE_HTML=test-output/cmd/csr/csr.html
if [ ! -f $FILE_HTML ]; then
  echo "ConflationStatsReportTest: html file not generated"
else
  echo "passed test 3"
fi

# TEST 4: simple file test to make sure pdf created
hoot conflate -D writer.include.debug=true -D stats.class="hoot::ScriptStatsComposer" -D  stats.script=./report/csr-default.js -D stats.format=pdf -D stats.output=test-output/cmd/csr/csr ./test-files/ToyTestA.osm ./test-files/ToyTestB.osm test-output/cmd/csr/csr2.osm --stats >& test-output/cmd/csr/csrStats.txt
FILE_PDF=test-output/cmd/csr/csr.pdf
if [ ! -f $FILE_PDF ]; then
  echo "ConflationStatsReportTest: pdf file not generated"
else
  echo "passed test 4"
fi

# TEST 5: test a sampling of the stats in the output of the stats command
# - will output difference if fails
python scripts/CheckThreeScores.py test-output/cmd/csr/csrStats.txt "Node Count,35,37,64,66,36,38" "Way Count,3,5,3,5,8,10" "Highway Count,3,5,3,5,8,10" "Total Unmatched Features,3,5,3,5,0,0"
echo "completed test 5"

# TEST 6: test all of the tokens used in the report.  make sure they are all there so someone cannot break the code
#   by renaming the stats tokens at random
# - will output difference if fails
python scripts/CheckTokens.py test-output/cmd/csr/csrStats.txt 'Node Count' 'Way Count' 'Relation Count' 'Total Feature Tags' 'Total Feature Information Tags' 'Total Feature Metadata Tags' 'Unique Names' 'Meters of Linear Features' 'Meters Squared of Area Features' 'POI Count' 'Highway Count' 'Meters of Highway' 'Highway Unique Name Count' 'Total Building Count' 'Meters Squared of Buildings' 'Building Unique Name Count' 'Total Feature Count' 'Total Conflated Features' 'Percentage of Total Features Conflated' 'Total Unmatched Features' 'Percentage of Total Features Unmatched' 'Conflated POIs' 'Conflated Highways' 'Meters of Conflated Highway' 'Conflated Buildings' 'Meters Squared of Conflated Buildings' 'Total Features Marked for Review' 'Percentage of Total Features Marked for Review' 'Longest Tag' 'Translated Populated Tag Percent' 'Translated Populated Tag Count' 'Translated Default Tag Count' 'Translated Null Tag Count' 'Building Translated Populated Tag Percent' 'Highway Translated Populated Tag Percent' 'POI Translated Populated Tag Percent' 'Read Inputs Time (sec)' 'Time to Calculate Stats for Input 1 (sec)' 'Time to Calculate Stats for Input 2 (sec)' 'Apply Named Ops Time (sec)' 'Apply Pre Ops Time (sec)' 'Project to Planar Time (sec)' 'Find Matches Time (sec)' 'Number of Matches Found' 'Number of Matches Found per Second' 'Number of Matches Before Whole Groups' 'Number of Whole Groups' 'Optimize Matches Time (sec)' 'Number of Optimized Matches' 'Number of Matches Optimized per Second' 'Create Mergers Time (sec)' 'Apply Mergers Time (sec)' 'Mergers Applied per Second' 'Apply Post Ops Time (sec)' 'Conflation Time (sec)' 'Project to WGS84 Time (sec)' 'Time to Calculate Stats for Output (sec)' 'Write Output Time (sec)' 'Total Time Elapsed (sec)' 'Bytes Read' 'Bytes Written'
