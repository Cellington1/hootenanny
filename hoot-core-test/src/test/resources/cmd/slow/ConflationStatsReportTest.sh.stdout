16:16:27.969 WARN  ...core/scoring/MapComparator.cpp( 233) Number of elements does not match.
447c447
<         <tag k="place" v="town"/>
---
>         <tag k="place" v="city"/>
575,576c575,576
<     <node visible="true" id="-296" timestamp="1970-01-01T00:00:00Z" version="1" lat="38.8543116509978290" lon="-104.8997171368439325">
<         <tag k="hoot:id" v="-296"/>
---
>     <node visible="true" id="-294" timestamp="1970-01-01T00:00:00Z" version="1" lat="38.8543116509978290" lon="-104.8997171368439325">
>         <tag k="hoot:id" v="-294"/>
579,580c579,580
<     <node visible="true" id="-298" timestamp="1970-01-01T00:00:00Z" version="1" lat="38.8532641860755490" lon="-104.8997533243950926">
<         <tag k="hoot:id" v="-298"/>
---
>     <node visible="true" id="-297" timestamp="1970-01-01T00:00:00Z" version="1" lat="38.8532641860755490" lon="-104.8997533243950926">
>         <tag k="hoot:id" v="-297"/>
583c583
<     <way visible="true" id="-252" timestamp="1970-01-01T00:00:00Z" version="1">
---
>     <way visible="true" id="-243" timestamp="1970-01-01T00:00:00Z" version="1">
607c607
<         <tag k="hoot:id" v="-252"/>
---
>         <tag k="hoot:id" v="-243"/>
610c610
<     <way visible="true" id="-250" timestamp="1970-01-01T00:00:00Z" version="1">
---
>     <way visible="true" id="-241" timestamp="1970-01-01T00:00:00Z" version="1">
617c617
<         <tag k="hoot:id" v="-250"/>
---
>         <tag k="hoot:id" v="-241"/>
620c620
<     <way visible="true" id="-247" timestamp="1970-01-01T00:00:00Z" version="1">
---
>     <way visible="true" id="-238" timestamp="1970-01-01T00:00:00Z" version="1">
627c627
<         <tag k="hoot:id" v="-247"/>
---
>         <tag k="hoot:id" v="-238"/>
630c630
<     <way visible="true" id="-243" timestamp="1970-01-01T00:00:00Z" version="1">
---
>     <way visible="true" id="-234" timestamp="1970-01-01T00:00:00Z" version="1">
636c636
<         <tag k="hoot:id" v="-243"/>
---
>         <tag k="hoot:id" v="-234"/>
639,640c639,640
<     <way visible="true" id="-240" timestamp="1970-01-01T00:00:00Z" version="1">
<         <nd ref="-298"/>
---
>     <way visible="true" id="-231" timestamp="1970-01-01T00:00:00Z" version="1">
>         <nd ref="-297"/>
663c663
<         <tag k="hoot:id" v="-240"/>
---
>         <tag k="hoot:id" v="-231"/>
666c666
<     <way visible="true" id="-236" timestamp="1970-01-01T00:00:00Z" version="1">
---
>     <way visible="true" id="-227" timestamp="1970-01-01T00:00:00Z" version="1">
671c671
<         <nd ref="-298"/>
---
>         <nd ref="-297"/>
675c675
<         <tag k="hoot:id" v="-236"/>
---
>         <tag k="hoot:id" v="-227"/>
678,687c678
<     <way visible="true" id="-233" timestamp="1970-01-01T00:00:00Z" version="1">
<         <nd ref="-296"/>
<         <nd ref="-20"/>
<         <tag k="note" v="0"/>
<         <tag k="highway" v="road"/>
<         <tag k="error:circular" v="15"/>
<         <tag k="hoot:id" v="-233"/>
<         <tag k="hoot:status" v="1"/>
<     </way>
<     <way visible="true" id="-232" timestamp="1970-01-01T00:00:00Z" version="1">
---
>     <way visible="true" id="-219" timestamp="1970-01-01T00:00:00Z" version="1">
690,711c681,682
<         <nd ref="-296"/>
<         <tag k="note" v="0"/>
<         <tag k="highway" v="road"/>
<         <tag k="error:circular" v="15"/>
<         <tag k="hoot:id" v="-232"/>
<         <tag k="hoot:status" v="3"/>
<     </way>
<     <way visible="true" id="-229" timestamp="1970-01-01T00:00:00Z" version="1">
<         <nd ref="-5"/>
<         <nd ref="-21"/>
<         <nd ref="-1"/>
<         <nd ref="-17"/>
<         <nd ref="-31"/>
<         <tag k="note" v="0"/>
<         <tag k="highway" v="road"/>
<         <tag k="error:circular" v="15"/>
<         <tag k="hoot:id" v="-229"/>
<         <tag k="hoot:status" v="3"/>
<     </way>
<     <way visible="true" id="-225" timestamp="1970-01-01T00:00:00Z" version="1">
<         <nd ref="-31"/>
<         <nd ref="-13"/>
---
>         <nd ref="-294"/>
>         <nd ref="-20"/>
715c686
<         <tag k="hoot:id" v="-225"/>
---
>         <tag k="hoot:id" v="-219"/>
765a737,757
>     <way visible="true" id="-216" timestamp="1970-01-01T00:00:00Z" version="1">
>         <nd ref="-5"/>
>         <nd ref="-21"/>
>         <nd ref="-1"/>
>         <nd ref="-17"/>
>         <nd ref="-31"/>
>         <tag k="note" v="0"/>
>         <tag k="highway" v="road"/>
>         <tag k="error:circular" v="15"/>
>         <tag k="hoot:id" v="-216"/>
>         <tag k="hoot:status" v="3"/>
>     </way>
>     <way visible="true" id="-212" timestamp="1970-01-01T00:00:00Z" version="1">
>         <nd ref="-31"/>
>         <nd ref="-13"/>
>         <tag k="note" v="0"/>
>         <tag k="highway" v="road"/>
>         <tag k="error:circular" v="15"/>
>         <tag k="hoot:id" v="-212"/>
>         <tag k="hoot:status" v="3"/>
>     </way>
932c924
<     <relation visible="true" id="-4" timestamp="1970-01-01T00:00:00Z" version="1">
---
>     <relation visible="true" id="-3" timestamp="1970-01-01T00:00:00Z" version="1">
935c927
<         <member type="way" ref="-252" role="outline"/>
---
>         <member type="way" ref="-243" role="outline"/>
943c935
<         <tag k="hoot:id" v="-4"/>
---
>         <tag k="hoot:id" v="-3"/>
